packages/frontend-v2/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,37 @@
+import { useState, Dispatch, SetStateAction } from "react";
+
+// NextJS executes server side code first, and then client side. Window object is only present client side,
+// so window could be undefined. This ensures the window object exists before we use it.
+const isWindow = typeof window !== "undefined";
+
+export function useLocalStorage<T>(
+  key: string,
```

Comment:
+1

Reply:
Decided not to do this because the hook already ensures that the key is always going to be correct. And since the plans key has an `id` field, it would have made things more complicated. Might end up doing this at a later time tho, but Aryan and I decided not to do this for this PR.

Reply:
imo storing any form of fetchable data in localstorage is not needed – it's more for the use case of tracking form progress (in onboarding, for example). and what do you mean by "the hook ensures that the key is always going to be correct?"?